### PR TITLE
Bottom Tab Navigator

### PR DESCRIPTION
**Pull request by mshago**
Blocker: waiting on the icons and the badge component

### CODE REVIEW

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <Icon/>
+      <Icon/>
+      <Icon/>
+      <Icon/>
+      <Icon/>
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
```
Comment (raynesmike) [inline mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx:31]: I think we need to separate the styles with a .ts extensions.

**Review by raynesmike — APPROVED** [review]
I approve this addition

**Review by carizzuti — APPROVED** [review]

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import {Icon} from '../icon/Icon';
+import { color, size } from '../../theme';
+
+const SIZE = size.lg;
+const PRIMARY_COLOR = color.primary;
+
+export const BottomTabNavigator: FC<BottomTabNavigatorProps> = ({}) => {
+
+  const handleOnPress = () => {
+    Alert.alert('Pressed')
```
Comment (lasharela) [inline mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx:15]: do we have to have this alert here?

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.tabContainer}>
+        <TouchableOpacity onPress={handleOnPress}>
+        <Icon name="Compass" height={SIZE} width={SIZE} fill={PRIMARY_COLOR} />
```
Comment (lasharela) [inline mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx:22]: why do we have local SIZE... some times it's needed. just asking. why not to set directly size something.

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.type.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export type BottomTabNavigatorProps = {
+  state?:string,
```
Comment (lasharela) [inline mobile/src/components/bottomTabNavigator/BottomTabNavigator.type.ts:2]: I can't figure out what state is about? whhy it needs to be provided from outside?

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.type.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export type BottomTabNavigatorProps = {
```
Comment (lasharela) [inline mobile/src/components/bottomTabNavigator/BottomTabNavigator.type.ts:1]: for types maybe we'll use TBottom...etc (T in front)

**Review by lasharela — CHANGES_REQUESTED** [review]

File: mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx (hunk 3 above)
Comment (mshago) [reply mobile/src/components/bottomTabNavigator/BottomTabNavigator.tsx:22]: In case the size is too big/small we can change it from there

**Comment by Dulaya** [comment]
* put marginTop: 'auto' inside your container. This will push your navbar to the bottom.
* You don't need the shadow because your navbar will always be on the bottom.
* Instead of using SIZE, make it a mandatory type and force the user to input iconSize.
   * in BottomTabNavigator.type.ts:
      export type TBottomTabNavigatorProps = {
         iconSize: number,
      }
* Optional: Allow user to pass color into each fill of the icons via props.
* Optional: Let the user put in the values as props but use default number. 
   Whoever will build the Home screen should be the one providing value.

**Comment by Dulaya** [comment]
* put marginTop: 'auto' inside your container. This will push your navbar to the bottom.
* You don't need the shadow because your navbar will always be on the bottom.
* Instead of using SIZE, make it a mandatory type and force the user to input iconSize.
   * in BottomTabNavigator.type.ts:
      export type TBottomTabNavigatorProps = {
         iconSize: number,
      }
* Optional: Allow user to pass color into each fill of the icons via props.
* Optional: Let the user put in the values as props but use default number. 
   Whoever will build the Home screen should be the one providing value.

**Review by Dulaya — APPROVED** [review]

**Pull request merged**